python/sglang/srt/layers/quantization/awq.py

```diff
@@ -394,16 +409,40 @@ def create_weights(
         layer.register_parameter("scales", scales)
 
     def process_weights_after_loading(self, layer: torch.nn.Module) -> None:
-        layer.qweight = torch.nn.Parameter(layer.qweight.data, requires_grad=False)
-        layer.qzeros = torch.nn.Parameter(layer.qzeros.data, requires_grad=False)
-        layer.scales = torch.nn.Parameter(layer.scales.data, requires_grad=False)
+        if _is_cpu:
+            assert (
+                _is_cpu_amx_available
+            ), "AWQLinearMethod on CPU requires that CPU has AMX support"
+            _amx_process_int4_packed_qweight_after_loading(
+                layer, ["qweight", "qzeros", "scales"], "awq"
+            )
+        else:
+            layer.qweight = torch.nn.Parameter(layer.qweight.data, requires_grad=False)
+            layer.qzeros = torch.nn.Parameter(layer.qzeros.data, requires_grad=False)
+            layer.scales = torch.nn.Parameter(layer.scales.data, requires_grad=False)
 
     def apply(
         self,
         layer: torch.nn.Module,
         x: torch.Tensor,
         bias: Optional[torch.Tensor] = None,
     ) -> torch.Tensor:
+        if use_intel_amx_backend(layer):
+            if SGLANG_USE_CPU_INT4_W4A8:
+                return torch.ops.sgl_kernel.int4_scaled_mm_cpu_with_quant(
+                    x,
+                    layer.qweight,
+                    layer.scales,
+                    layer.qzeros,
+                    layer.compensation,
+                    bias,
+                    x.dtype,
+                )
+            else:
+                return torch.ops.sgl_kernel.int4_scaled_mm_cpu(
+                    x, layer.qweight, layer.qzeros, layer.scales, bias
+                )
```

Comment:
use the same kernel and pack compensation with qweight, just like int8 path.

Reply:
Yes, have refine the frontend here, and will make the kernel changes in https://github.com/sgl-project/sglang/pull/8226